src/cursor.c

```diff
@@ -2098,8 +2099,9 @@ if (0) fprintf(stderr, "check_x11_pointer %d %d\n", root_x, root_y);
 				if (debug_pointer) {
 					rfbLog("Skipping cursor_position() outside our clipshift\n");
 				}
-				return 0;
 			}
+			rint = set_cursor(0, 0, CURS_EMPTY);
```

Comment:
What if it was another cursor previously?

Reply:
Cursor was another previously, but it was inside clipped area. Now cursor out of it so it must be hidden. 
When cursor returns to clipped area then [line 2112](https://github.com/LibVNC/x11vnc/pull/149/files/af8fed5cf38661bf326f3d7afa4f625ae2160997#diff-96a84fd451ccce58565901b35bf95e79d7bb380e980d827fae657591d48b925bR2112) will be called and cursor will be restored to correct state